cfme/tests/services/test_rest_services.py

```diff
@@ -171,6 +171,21 @@ def vm(request, a_provider, appliance):
     return _vm(request, a_provider, appliance.rest_api)
 
 
+@pytest.yield_fixture(scope="function")
+def cart(appliance):
+    cart = appliance.rest_api.collections.service_orders.action.create(name="cart")
+    assert_response(appliance)
+    cart = cart[0]
+
+    yield cart
+
+    try:
+        cart.action.delete()
```

Comment:
**REQUIRED**: We need to add a finalizer to the relevant tests, else the ones that don't may fail during delete and we'd not know

Reply:
@psav I changed it so the exception handling doesn't mask delete failures. For the majority of tests simply deleting the cart when it's present is enough. In tests where ``delete`` actions are tested there's checking for status codes etc. to make sure the delete actions behaves as expected.